pkg/webhooks/prometheusrule/prometheusrule_test.go

```diff
@@ -0,0 +1,279 @@
+			targetNamespace: "openshift-monitoring",
+			targetResource:  "prometheusrule",
+			username:        "system:serviceaccount:openshift-monitoring:prometheus-operator",
+			userGroups:      []string{"system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Delete,
+			shouldBeAllowed: true,
+		},
+		{
+
+			testID:          "Allowed-can-update-prometheusrule-in-managed-namespaces",
+			targetNamespace: "openshift-monitoring",
+			username:        "system:serviceaccount:openshift-monitoring:prometheus-operator",
+			targetResource:  "prometheusrule",
+			userGroups:      []string{"system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Update,
+			shouldBeAllowed: true,
+		},
+		{
+
+			testID:          "admin-can-create-prometheusrule-in-managed-namespaces",
+			targetNamespace: "openshift-monitoring",
+			username:        "kube:admin",
+			targetResource:  "prometheusrule",
+			userGroups:      []string{"system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Create,
+			shouldBeAllowed: true,
+		},
+		{
+
+			testID:          "admin-can-create-prometheusrule-in-managed-namespaces",
+			targetNamespace: "openshift-monitoring",
+			username:        "kube:admin",
+			targetResource:  "prometheusrule",
+			userGroups:      []string{"system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Update,
+			shouldBeAllowed: true,
+		},
+		{
+
+			testID:          "admin-can-create-prometheusrule-in-managed-namespaces",
+			targetNamespace: "openshift-monitoring",
+			username:        "kube:admin",
+			targetResource:  "prometheusrule",
+			userGroups:      []string{"system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Delete,
+			shouldBeAllowed: true,
+		},
+		{
+			testID:          "regular-user-can-create-prometheusrule-in-managed-namespaces",
+			targetNamespace: "openshift-user-workload-monitoring",
+			targetResource:  "prometheusrule",
+			username:        "user1",
+			userGroups:      []string{"system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Create,
+			shouldBeAllowed: true,
+		},
+		{
+			testID:          "regular-user-can-create-prometheusrule-in-managed-namespaces",
+			targetNamespace: "openshift-user-workload-monitoring",
+			targetResource:  "prometheusrule",
+			username:        "user2",
+			userGroups:      []string{"system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Delete,
+			shouldBeAllowed: true,
+		},
+		{
+			testID:          "regular-user-can-create-prometheusrule-in-managed-namespaces",
+			targetNamespace: "openshift-user-workload-monitoring",
+			targetResource:  "prometheusrule",
+			username:        "user3",
+			userGroups:      []string{"system:authenticated", "system:authenticated:oauth"},
+			operation:       admissionv1.Update,
+			shouldBeAllowed: true,
```

Comment:
what are the difference between these 3 test cases?  the testID should mean the user can create prometheusrule in openshift-user-workload-monitoring but not other managed namespaces?

Reply:
yes